### DISCUSSION TITLE
Returning bits per channel?

### DISCUSSION THREAD

**Discussion by marauder2k7** [nothings/stb#1527] [Q&A] [answered]
Trying to implement stb image for loading and writing files, was curious if there is a way to return the bits per channel? 

in our engine we support 16bpc and 32bpc textures for some very specific cases. I can see that there is a specific 16bpc load from memory but i can't see how to be able to tell programmatically from the documentation as to how to tell when that should be used?

**Comment by nothings** [comment, marked as answer]
`stbi_is_16_bit` and `stbi_is_16_bit_from_memory`.

We don't support 32-bit except float.

**Reply by marauder2k7** [reply]
nice that should sort the issue 👍 

is there also a way to tell whether it should be float? (we can do without the float one tbh, i think most of the time this will be an internally created texture, but if we can find out if its supposed to be read this way then it would expand usage

**Reply by marauder2k7** [reply]
i can see if its float its always treated as hdr so i can just do the stbi_is_hdr_from_memory call thanks for your help :) 👍
